gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -73,18 +71,17 @@ def run(
 
         # convert translation direction in global frame using rotations.
         w_i2Ui1_measurements = BinaryMeasurementsUnit3()
+        num_valid_measurements = 0
         for (i1, i2), i2Ui1 in i2Ui1_dict.items():
             if i2Ui1 is not None and wRi_list[i2] is not None:
                 w_i2Ui1_measurements.append(
                     BinaryMeasurementUnit3(i2, i1, Unit3(wRi_list[i2].rotate(i2Ui1.point3())), noise_model)
                 )
+                num_valid_measurements += 1
 
         # sample indices to be used as projection directions
-        num_measurements = len(i2Ui1_dict)
         indices = np.random.choice(
-            num_measurements,
-            min(self._max_1dsfm_projection_directions, num_measurements),
-            replace=False,
+            num_valid_measurements, min(self._max_1dsfm_projection_directions, num_valid_measurements), replace=False,
```

Comment:
i think a cleaner fix is
```python
        # sample indices to be used as projection directions
        # note: if some unit translations are None, then we can only use valid values
        num_measurements = len(w_i2Ui1_measurements)
        indices = np.random.choice(
            num_measurements,
            size=min(self._max_1dsfm_projection_directions, num_measurements),
            replace=False,
        )
```

Reply:
do you know why the unit translations end up being None here? does MFAS mark them as outliers or something?

Reply:
changed the logic. I do not think we need the comments though.

No its because rotation averaging might return some None's. Or the input relative directions might itself be None.